tests/x11/yast2_lan_restart.pm

```diff
@@ -0,0 +1,311 @@
+}
+
+sub check_network {
+    my $status = shift || 'no_restart';
+    wait_still_screen;
+    send_key 'alt-o';                           # OK
+    assert_screen 'yast2_closed_xterm_visible', 120;
+    script_run 'ip a';
+    # workaround bsc#992113
+    #assert_script_run 'dig suse.com|grep status';    # test if conection and DNS is working
+    if ("$status" eq 'restart') {
+        assert_script_run '[ -s strace.log ]';    # strace.log size is greater than zero (network restarted)
+    }
+    else {
+        script_run 'cat strace.log';                # print strace.log
+        assert_script_run '[ ! -s strace.log ]';    # strace.log size is not greater than zero (network not restarted)
+    }
+    script_run "cat strace.log > /dev/$serialdev";    # print strace.log
+    script_run 'cat /dev/null > strace.log';          # clear strace.log
+    type_string "\n\n";                               # make space for better readability of the console
+}
+
+sub add_device {
+    my $device = shift;
+    assert_screen 'yast2_closed_xterm_visible', 120;
+    run_yast2_lan;
+    if ("$device" eq 'bond') {
+        send_key 'alt-i';                             # Edit NIC
+        assert_screen 'yast2_lan_network_card_setup';
+        send_key 'alt-k';                             # No link (Bonding Slavees)
+        send_key 'alt-n';                             # Next
+        assert_screen 'yast2_lan';                    # yast2 lan overview tab
+    }
+    send_key 'alt-a';                                 # Add NIC
+    assert_screen 'yast2_lan_hardware_dialog';
+    send_key 'alt-d';                                 # Device type
+    send_key 'home';                                  # Jump to beginning of list
+    send_key_until_needlematch "yast2_lan_device_type_$device", 'down';
+    send_key 'alt-n';                                 # Next
+    assert_screen 'yast2_lan_network_card_setup';
+    send_key 'alt-y';                                 # Dynamic address
+    wait_still_screen;
+    if ("$device" eq 'bridge') {
+        send_key 'alt-g';                             # General
+        send_key 'alt-i';                             # Bridged devices
+        assert_screen 'yast2_lan_bridged_devices';
+        if (check_screen('yast2_lan_default_NIC_bridge')) {
+            send_key 'alt-d';                         # select Bridged Devices region
+            send_key 'spc';
+            wait_still_screen;
+            save_screenshot;
+        }
+        send_key 'alt-n';                             # Next
+        assert_screen 'yast2_lan_select_already_configured_device';
+        send_key 'alt-o';                             # OK
+    }
+    elsif ("$device" eq 'bond') {
+        send_key 'alt-o';                             # Bond slaves
+        assert_screen 'yast2_lan_bond_slaves';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
+        send_key 'tab';
```

Comment:
I expected this ;) but I'm not sure how robust is to match blinking cursor in text field